src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Internal/PrecompileCommandBase.cs

```diff
@@ -0,0 +1,184 @@
+
+        protected CommandOption ConfigurationOption { get; set; }
+
+        protected CommandOption ConfigureCompilationType { get; set; }
+
+        protected CommandOption ContentRootOption { get; set; }
+
+        protected CommandOption GeneratePdbOption { get; set; }
+
+        protected string ProjectPath { get; private set; }
+
+        protected NuGetFramework TargetFramework { get; private set; }
+
+        protected string Configuration { get; private set; }
+
+        public static void Register<TPrecompileCommandBase>(CommandLineApplication app) where
+            TPrecompileCommandBase : PrecompileCommandBase, new()
+        {
+            var command = new TPrecompileCommandBase();
+            command.Configure(app);
+        }
+
+        protected virtual void Configure(CommandLineApplication app)
+        {
+            app.Description = "Precompiles an application.";
+            app.HelpOption("-?|-h|--help");
+            ProjectArgument = app.Argument(
+                "project",
+                "The path to the project (project folder or project.json) with precompilation.");
+            FrameworkOption = app.Option(
+                "-f|--framework",
+                "Target Framework",
+                CommandOptionType.SingleValue);
+            OutputPathOption = app.Option(
+                "-o|--output",
+                "The output (bin or publish) directory.",
+                CommandOptionType.SingleValue);
+
+            ConfigurationOption = app.Option(
+                "-c|--configuration",
+                "Configuration",
+                CommandOptionType.SingleValue);
+
+            ConfigureCompilationType = app.Option(
+                "--configure-compilation-type",
+                "Type with Configure method",
+                CommandOptionType.SingleValue);
+
+            ContentRootOption = app.Option(
+                "--content-root",
+                "The application's content root.",
+                CommandOptionType.SingleValue);
+
+            GeneratePdbOption = app.Option(
+                "--generate-pdbs",
+                "Generate pdbs for views.",
+                CommandOptionType.NoValue);
+
+            app.OnExecute((Func<int>)Execute);
+        }
+
+        protected int Execute()
+        {
+            if (!ParseArguments())
+            {
+                return 1;
+            }
+
+            return ExecuteCore();
+        }
+
+        protected abstract int ExecuteCore();
+
+        private bool ParseArguments()
+        {
+            if (!string.IsNullOrEmpty(ProjectArgument.Value))
+            {
+                ProjectPath = Path.GetFullPath(ProjectArgument.Value);
+                if (string.Equals(Path.GetFileName(ProjectPath), "project.json", StringComparison.OrdinalIgnoreCase))
+                {
+                    ProjectPath = Path.GetDirectoryName(ProjectPath);
+                }
+
+                if (!Directory.Exists(ProjectPath))
```

Comment:
Isn't this only possible if someone chooses to call the inside-man manually? I'd think the .Tools package would explode of the project.json wasn't valid.